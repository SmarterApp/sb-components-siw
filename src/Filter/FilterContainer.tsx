import * as React from "react";
import {
  AdvancedFilterContainer,
  BasicFilterContainer,
  AdvancedFilterCategoryModel,
  BasicFilterCategoryModel,
  FilterOptionModel,
  FilterType
} from "@src/index";

export interface FilterContainerProps {
  basicFilterCategories: BasicFilterCategoryModel[];
  onUpdateBasicFilter: (
    selected: BasicFilterCategoryModel[],
    changed: FilterType
  ) => void;
  advancedFilterCategories: AdvancedFilterCategoryModel[];
  onUpdateAdvancedFilter: (
    selected: AdvancedFilterCategoryModel[],
    changed?: FilterType
  ) => void;
  filterId?: string;
}

export interface FilterContainerState {
  expanded: boolean;
}

export class FilterContainer extends React.Component<
  FilterContainerProps,
  FilterContainerState
> {
  constructor(props: FilterContainerProps) {
    super(props);
    this.state = {
      expanded: false
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({ expanded: !this.state.expanded });
  }

  render() {
    const {
      basicFilterCategories,
      onUpdateBasicFilter,
      advancedFilterCategories,
      onUpdateAdvancedFilter,
      filterId
    } = this.props;
    const id = filterId || "";
    let advancedFilter: JSX.Element | undefined;
    if (this.state.expanded) {
      advancedFilter = (
        <AdvancedFilterContainer
          isNested={true}
          filterCategories={advancedFilterCategories}
          onUpdateFilter={onUpdateAdvancedFilter}
        />
      );
    }

    return (
      <div className="filter-component-wrapper">
        <BasicFilterContainer
          filterId={filterId}
          filterCategories={basicFilterCategories}
          onUpdateFilter={onUpdateBasicFilter}
          containsAdvancedFilter={true}
          handleAdvancedFilterExpand={this.handleClick}
        />
        {advancedFilter}
      </div>
    );
  }
}
