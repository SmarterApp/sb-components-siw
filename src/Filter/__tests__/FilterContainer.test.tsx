import * as React from "react";
import { FilterContainer } from "../FilterContainer";
import {
  BasicFilterCategoryModel,
  AdvancedFilterCategoryModel
} from "../FilterModels";
import {
  mockBasicFilterCategories,
  mockAdvancedFilterCategoriesAll
} from "../../../mocks/Filter/mocks";
import { shallow } from "enzyme";
import { BasicFilterContainer } from "../BasicFilterContainer";

describe("FilterContainer", () => {
  const updateBasicFilter = (
    basicFilterCategories: BasicFilterCategoryModel[]
  ) => {
    wrapper.setProps({ basicFilterCategories });
  };

  const updateAdvancedFilter = (
    advancedFilterCategories: AdvancedFilterCategoryModel[]
  ) => {
    wrapper.setProps({ advancedFilterCategories });
  };

  const wrapper = shallow(
    <FilterContainer
      basicFilterCategories={mockBasicFilterCategories}
      onUpdateBasicFilter={updateBasicFilter}
      advancedFilterCategories={mockAdvancedFilterCategoriesAll}
      onUpdateAdvancedFilter={updateAdvancedFilter}
    />,
    { lifecycleExperimental: true }
  );

  it("expands the advanced filter", () => {
    expect(wrapper).toMatchSnapshot();
    for (let i = 0; i < 2; i++) {
      wrapper
        .findWhere(node => node.type() === BasicFilterContainer)
        .dive()
        .find(".af-expand")
        .simulate("click");
      expect(wrapper).toMatchSnapshot();
    }
  });
});
