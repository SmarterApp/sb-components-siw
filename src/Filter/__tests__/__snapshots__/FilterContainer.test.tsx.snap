// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterContainer expands the advanced filter 1`] = `
<div
  className="filter-component-wrapper"
>
  <BasicFilterContainer
    containsAdvancedFilter={true}
    filterCategories={
      Array [
        Object {
          "code": "Grade",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "7",
              "label": "Grades 3-5",
            },
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "56",
              "label": "Grades 6-8",
            },
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "960",
              "label": "High School",
            },
          ],
          "label": "Grade",
          "optionType": 3,
          "show": true,
        },
        Object {
          "code": "Subject",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "Subject",
              "isSelected": false,
              "key": "ELA",
              "label": "English Literacy/ Arts",
            },
            Object {
              "filterType": "Subject",
              "isSelected": false,
              "key": "Math",
              "label": "Mathematics",
            },
          ],
          "label": "Subjects",
          "optionType": 2,
          "show": true,
        },
        Object {
          "code": "TechnologyType",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "TechnologyType",
              "isSelected": false,
              "key": "tech1",
              "label": "tech1",
            },
            Object {
              "filterType": "TechnologyType",
              "isSelected": false,
              "key": "tech2",
              "label": "tech2",
            },
          ],
          "label": "TechType",
          "optionType": 4,
          "show": true,
        },
      ]
    }
    handleAdvancedFilterExpand={[Function]}
    onUpdateFilter={[Function]}
  />
</div>
`;

exports[`FilterContainer expands the advanced filter 2`] = `
<div
  className="filter-component-wrapper"
>
  <BasicFilterContainer
    containsAdvancedFilter={true}
    filterCategories={
      Array [
        Object {
          "code": "Grade",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "7",
              "label": "Grades 3-5",
            },
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "56",
              "label": "Grades 6-8",
            },
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "960",
              "label": "High School",
            },
          ],
          "label": "Grade",
          "optionType": 3,
          "show": true,
        },
        Object {
          "code": "Subject",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "Subject",
              "isSelected": false,
              "key": "ELA",
              "label": "English Literacy/ Arts",
            },
            Object {
              "filterType": "Subject",
              "isSelected": false,
              "key": "Math",
              "label": "Mathematics",
            },
          ],
          "label": "Subjects",
          "optionType": 2,
          "show": true,
        },
        Object {
          "code": "TechnologyType",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "TechnologyType",
              "isSelected": false,
              "key": "tech1",
              "label": "tech1",
            },
            Object {
              "filterType": "TechnologyType",
              "isSelected": false,
              "key": "tech2",
              "label": "tech2",
            },
          ],
          "label": "TechType",
          "optionType": 4,
          "show": true,
        },
      ]
    }
    handleAdvancedFilterExpand={[Function]}
    onUpdateFilter={[Function]}
  />
</div>
`;

exports[`FilterContainer expands the advanced filter 3`] = `
<div
  className="filter-component-wrapper"
>
  <BasicFilterContainer
    containsAdvancedFilter={true}
    filterCategories={
      Array [
        Object {
          "code": "Grade",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "7",
              "label": "Grades 3-5",
            },
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "56",
              "label": "Grades 6-8",
            },
            Object {
              "filterType": "Grade",
              "isSelected": false,
              "key": "960",
              "label": "High School",
            },
          ],
          "label": "Grade",
          "optionType": 3,
          "show": true,
        },
        Object {
          "code": "Subject",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "Subject",
              "isSelected": false,
              "key": "ELA",
              "label": "English Literacy/ Arts",
            },
            Object {
              "filterType": "Subject",
              "isSelected": false,
              "key": "Math",
              "label": "Mathematics",
            },
          ],
          "label": "Subjects",
          "optionType": 2,
          "show": true,
        },
        Object {
          "code": "TechnologyType",
          "disabled": false,
          "filterOptions": Array [
            Object {
              "filterType": "TechnologyType",
              "isSelected": false,
              "key": "tech1",
              "label": "tech1",
            },
            Object {
              "filterType": "TechnologyType",
              "isSelected": false,
              "key": "tech2",
              "label": "tech2",
            },
          ],
          "label": "TechType",
          "optionType": 4,
          "show": true,
        },
      ]
    }
    handleAdvancedFilterExpand={[Function]}
    onUpdateFilter={[Function]}
  />
</div>
`;
